pkg/cosign/remote.go

```diff
@@ -52,15 +52,23 @@ func Descriptors(ref name.Reference) ([]v1.Descriptor, error) {
 	return m.Layers, nil
 }
 
-// SignatureImage
+// SignatureImage returns the existing destination image, or a new, empty one.
 func SignatureImage(dstTag name.Reference, opts ...remote.Option) (v1.Image, error) {
 	base, err := remote.Image(dstTag, opts...)
 	if err != nil {
 		if te, ok := err.(*transport.Error); ok {
 			if te.StatusCode != http.StatusNotFound {
 				return nil, te
 			}
-			base = empty.Image
+			if !LegacyMediaType() {
+				base = mutate.MediaType(empty.Image, types.OCIManifestSchema1)
+				m, err := base.Manifest()
+				if err != nil {
+					// should never happen...?
+					return nil, err
+				}
+				m.Config.MediaType = types.OCIConfigJSON
```

Comment:
Oh that's interesting. I don't think we intended for that return value to be mutable, but I guess it is...

cc @imjasonh I would consider this a bug.

Reply:
Hahha.

Reply:
let us have our fun :p